src/extender.rs

```diff
@@ -93,24 +93,24 @@ impl<H: Hasher> Extender<H> {
         }
     }
 
-    fn add_unit(&mut self, u: ExtenderUnit<H>) -> bool {
-        debug!(target: "rush-extender", "{} New unit in Extender {:?}.", self.node_id, u.hash);
+    fn add_unit(&mut self, u: ExtenderUnit<H>) {
+        debug!(target: "rush-extender", "{} New unit in Extender round {} creator {} hash {:?}.", self.node_id, u.round, u.creator, u.hash);
         let round = u.round;
         if round > self.state.highest_round {
             self.state.highest_round = round;
         }
-        debug!(target: "rush-extender", "{} unit round {:?} state current_round {:?}", self.node_id, u.round, self.state.current_round);
-        // need to extend the vector first to the required length
-        if self.units_by_round.len() <= round {
-            self.units_by_round.push(vec![]);
-        }
+
         if round >= self.state.current_round {
+            //The units in units_by_round are required for head calculation only.
+            //If the round of u is too low, we don't need to update it.
+
+            //Need to extend the vector first to the required length.
+            if self.units_by_round.len() <= round {
+                self.units_by_round.push(vec![]);
+            }
             self.units_by_round[round].push(u.hash);
-            self.units.insert(u.hash, u);
-            true
-        } else {
-            false
         }
+        self.units.insert(u.hash, u);
```

Comment:
There is a comment just below `if round >= self.state.current_round ` in which I'm trying to explain that. Essentially, units_by_round are requires to calculate heads, so 
1) if we have a unit that is below the current round, we don't need to add it to this structure
2) but still, even if the unit is below the current round it will be at some point part of some batch, hence we cannot just skip it :)

Reply:
The comment and point 1. is clear, my concern is point 2. Will it be a part of some batch in current version? We don't connect to old units.

Reply:
Yes, it is possible, for instance a guy who slept starts when 10 rounds are already there. Then he will catch up by round say 12, and in round 13 or so all its units will be in the batch.

Reply:
That's right :)